Core/WebServer.cpp

```diff
@@ -147,19 +165,30 @@ static std::string RemotePathForRecent(const std::string &filename) {
 	// Let's not serve directories, since they won't work.  Only single files.
 	// Maybe can do PBPs and other files later.  Would be neat to stream virtual disc filesystems.
 	if (RemoteISOFileSupported(basename)) {
-		return ReplaceAll(basename, " ", "%20");
+		return ServerUriEncode(basename);
 	}
-	return "";
+
+	return std::string();
 }
 
 static Path LocalFromRemotePath(const std::string &path) {
-	for (const std::string &filename : g_Config.RecentIsos()) {
-		std::string basename = RemotePathForRecent(filename);
-		if (basename == path) {
-			return Path(filename);
+	switch ((RemoteISOShareType)g_Config.iRemoteISOShareType) {
+	case RemoteISOShareType::RECENT:
+		for (const std::string &filename : g_Config.RecentIsos()) {
+			std::string basename = RemotePathForRecent(filename);
+			if (basename == path) {
+				return Path(filename);
+			}
 		}
+		return Path();
+	case RemoteISOShareType::LOCAL_FOLDER:
+	{
+		std::string decoded = ServerUriDecode(path);
+		return Path(g_Config.sRemoteISOSharedDir) / decoded;
```

Comment:
This is probably technically a security gap.  The Recent path is intentionally a bit careful to only take the filename-part, and not take for example "../../../passwd".  I guess this is probably fine as hopefully you're only on a network with devices you trust...

-[Unknown]

Reply:
Yeah... Probably worth adding checks against though indeed.